content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-12.md

```diff
@@ -0,0 +1,289 @@
+    [bootstraptoken] configured RBAC rules to allow certificate rotation for all node client certificates in the cluster
+    [addons] Applied essential addon: CoreDNS
+    [addons] Applied essential addon: kube-proxy
+
+    [upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.12.0-rc.1". Enjoy!
+
+    [upgrade/kubelet] Now that your control plane is upgraded, please proceed with upgrading your kubelets if you haven't already done so.
+    ```
+
+1.  Manually upgrade your Software Defined Network (SDN).
+
+    Your Container Network Interface (CNI) provider may have its own upgrade instructions to follow.
+    Check the [addons](/docs/concepts/cluster-administration/addons/) page to
+    find your CNI provider and see whther additional upgrade steps are required.
+
+## Upgrade master and node packages
+
+1.  Prepare each host for maintenance, marking it unschedulable and evicting the workload:
+
+    ```shell
+    kubectl drain $HOST --ignore-daemonsets
+    ```
+
+    On the master host, you must add `--ignore-daemonsets`:
+
+    ```shell
+    kubectl drain ip-172-31-85-18
+    node "ip-172-31-85-18" cordoned
+    error: unable to drain node "ip-172-31-85-18", aborting command...
+
+    There are pending nodes to be drained:
+    ip-172-31-85-18
+    error: DaemonSet-managed pods (use --ignore-daemonsets to ignore): calico-node-5798d, kube-proxy-thjp9
+    ```
+
+    ```
+    kubectl drain ip-172-31-85-18 --ignore-daemonsets
+    node "ip-172-31-85-18" already cordoned
+    WARNING: Ignoring DaemonSet-managed pods: calico-node-5798d, kube-proxy-thjp9
+    node "ip-172-31-85-18" drained
+    ```
+
+1.  Upgrade the Kubernetes package version on each `$HOST` node by running the Linux package manager for your distribution:
+
+    {{< tabs name="k8s_install" >}}
+    {{% tab name="Ubuntu, Debian or HypriotOS" %}}
+    apt-get update
+    apt-get upgrade -y kubelet kubeadm
+    {{% /tab %}}
+    {{% tab name="CentOS, RHEL or Fedora" %}}
+    yum upgrade -y kubelet kubeadm --disableexcludes=kubernetes
+    {{% /tab %}}
+    {{< /tabs >}}
+
+## Upgrade kubelet on each node
+
+1.  On each node except the master node, upgrade the kubelet config:
+
+    ```shell
+    sudo kubeadm upgrade node config --kubelet-version $(kubelet --version | cut -d ' ' -f 2)
+    ```
+
+1.  Restart the kubectl process:
+
+    ```shell
+    sudo systemctl restart kubelet
+    ```
+
+1.  Verify that the new version of the `kubelet` is running on the host:
+
+    ```shell
+    systemctl status kubelet
+    ```
+
+1.  Bring the host back online by marking it schedulable:
+
+    ```shell
+    kubectl uncordon $HOST
+    ```
+
+1.  After the kubelet is upgraded on all hosts, verify that all nodes are available again by running the following command from anywhere -- for example, from outside the cluster:
+
+    ```shell
+    kubectl get nodes
+    ```
+
+    The `STATUS` column should show `Ready` for all your hosts, and the version number should be updated.
```

Comment:
I'm not sure whether we want to use `hosts` or `nodes` here and some of the other places in this doc, but I think `nodes` might be more exact in some places.